include/stdexec/__detail/__basic_sender.hpp

```diff
@@ -595,6 +599,9 @@ namespace stdexec {
 
   //////////////////////////////////////////////////////////////////////////////////////////////////
   // __make_sexpr
+  //! A tagged function-object
+  //! Takes data and children and 
+  //! returns `__sexpr_t<_Tag, _Data, _Child...>{_Tag(), data, children...}`.
```

Comment:
I didn't quite grok this `sexpr` stuff... Are there docs?

Reply:
Regarding "docs", personally, I find comments like the ones I'm trying to add in this PR to be more useful than "docs", since go-to-definition lands on them directly. I'll try to paraphrase what you are saying here.

Reply:
Do you mean `basic-sender` is represented by `__sexpr`? Because it doesn't seem to match... Plus there's `__basic_sender` defined on lines I touched above. Which is which?

Reply:
`stdexec::__basic_sender` is a dummy type used only for diagnostic purposes. `__sexpr` is the implementation of P2300's _`basic-sender`_.

Reply:
Ok. Want me to rename things or in this PR just document that fact?

Reply:
Also, _`basic-sender`_ is
```cpp
  template<class Tag, class Data, class... Child>
  struct basic-sender : product-type<Tag, Data, Child...> {    // exposition only
    using sender_concept = sender_t;
```
but `__sexpr` is
```cpp
  template <auto _DescriptorFn, class = __anon>
  struct __sexpr {
    using sender_concept = sender_t;
```
so something doesn't make sense... I assume you are being imprecise and that something related to `__sexpr` corresponds to _`basic-sender`_?

Reply:
i didn't mean to imply that `__sexpr` is exact clone of _`basic-sender`_. it does some things differently, but it largely follows the design of _`basic-sender`_.

no need to rename things. i like them as they are.

Reply:
Ok, if it isn’t quite _`basic-sender`_ I’ll say it’s similar to it. But I did comment that `__basic_sender` is not in fact providing _`basic-sender`_ which seems important to explicitly state.